powerline/segments/common/env.py

```diff
@@ -19,20 +19,23 @@ def environment(pl, segment_info, variable=None):
 
 
 @requires_segment_info
-def virtualenv(pl, segment_info, ignore_venv=False, ignore_conda=False):
+def virtualenv(pl, segment_info, ignore_venv=False, ignore_conda=False, ignored_names=("venv", ".venv")):
 	'''Return the name of the current Python or conda virtualenv.
-
+	:param Iterable ignored_names:
+		Names of venvs to ignore. Will then get the name of the venv by ascending to the parent directory
 	:param bool ignore_venv:
 		Whether to ignore virtual environments. Default is False.
 	:param bool ignore_conda:
 		Whether to ignore conda environments. Default is False.
 	'''
-	return (
-		(not ignore_venv and
-		 os.path.basename(segment_info['environ'].get('VIRTUAL_ENV', ''))) or
-		(not ignore_conda and
-		 segment_info['environ'].get('CONDA_DEFAULT_ENV', '')) or
-		None)
+	if not ignore_venv:
+		for candidate in reversed(segment_info['environ'].get('VIRTUAL_ENV', '').split("/")):
+			if candidate and candidate not in ignored_names:
+				return candidate
+	if not ignore_conda:
+		for candidate in reversed(segment_info['environ'].get('CONDA_DEFAULT_ENV', '').split("/")):
+			if candidate and candidate not in ignored_names:
+				return candidate
 
```

Comment:
Also, I think it would be more in line with the rest of the code to explicitly return `None` here.

Reply:
I thought that was kind of redundant, but I can add that

Reply:
While that is true, I think adding an explicit return statement makes the code more readable, as it pronounces the end of that function. (And even if it is redundant, it is only a single line, so it should be fine.)